InvoiceReader.Application/Queries/GetInvoices/Model.cs

```diff
@@ -9,9 +10,34 @@ public record Result
         {
             public List<Invoice> Data { get; set; }
 
-            public Result()
+            public Result(List<InvoiceItem> invoiceItems)
             {
-                // Required by Auto Mapper
+                if (invoiceItems != null)
+                {
+                    Data = new List<Invoice>();
+
+                    foreach (var invoiceItem in invoiceItems)
+                    {
+                        var invoice = new Invoice
+                        (
+                            id: invoiceItem.Id,
+                            referenceId: invoiceItem.Attributes.ReferenceId,
+                            creditorReference: invoiceItem.Attributes.CreditorReference,
+                            currency: invoiceItem.Attributes.Currency,
+                            net: invoiceItem.Attributes.NetAmount,
+                            gross: invoiceItem.Attributes.GrossAmount,
+                            remainder: invoiceItem.Attributes.Remainder,
+                            vat: invoiceItem.Attributes.VatAmount,
+                            expireDate: invoiceItem.Attributes.InvoiceExpirationDate,
+                            dueDate: invoiceItem.Attributes.DueDate,
+                            issueDate: invoiceItem.Attributes.Date,
+                            createdOn: invoiceItem.Attributes.DateCreated,
+                            updatedOn: invoiceItem.Attributes.DateUpdated
+                        );
+
+                        Data.Add(invoice);
+                    }
+                }
             }
```

Comment:
Our record should not deal with InvoiceItem.
It should accept List<Invoice> as an argument. And assign that accordingly.